### PR TITLE
Refactor RoutePrefix to be a string

### PR DESCRIPTION
**Pull request by nwoltman**
It used to be a class with a single `prefix` property that didn't have any extra functionality.
The prefix property as a string is all that's needed.

<details>
<summary>benchmark</summary>

**master**
```
[1] Stat         Avg     Stdev  Max
[1] Latency (ms) 3.11    11.01  360
[1] Req/Sec      31406.4 6013.7 35647
[1] Bytes/Sec    4.73 MB 897 kB 5.51 MB
[1]
[1] 157k requests in 5s, 23.4 MB read
...
[1] Stat         Avg     Stdev   Max
[1] Latency (ms) 3.01    10.83   371
[1] Req/Sec      32379.2 6412.03 37567
[1] Bytes/Sec    4.84 MB 964 kB  5.77 MB
[1]
[1] 162k requests in 5s, 24.1 MB read
```

**this PR**
```
[1] Stat         Avg    Stdev  Max
[1] Latency (ms) 3.16   11.16  371
[1] Req/Sec      30880  6228   35999
[1] Bytes/Sec    4.6 MB 912 kB 5.51 MB
[1]
[1] 154k requests in 5s, 23 MB read
...
[1] Stat         Avg     Stdev   Max
[1] Latency (ms) 3.01    10.7    373
[1] Req/Sec      32462.4 6622.57 36767
[1] Bytes/Sec    4.81 MB 1 MB    5.51 MB
[1]
[1] 162k requests in 5s, 24.2 MB read
```
</details>

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jsumners** [comment]
I wonder if the object was there due to V8 classes.

**Comment by nwoltman** [comment]
Something about keeping the same shape of the fastify instance?

**Comment by allevo** [comment]
Could you explain why of this PR?

**Comment by nwoltman** [comment]
Makes the code easier to understand and maintain.

File: fastify.js [hunk 1]
```diff
@@ -313,21 +313,15 @@ function build (options) {
...
-    if (typeof opts.prefix === 'string') {
-      if (opts.prefix[0] !== '/') {
-        opts.prefix = '/' + opts.prefix
-      }
-      R.prefix += opts.prefix
+    if (pluginPrefix[0] !== '/') {
+      pluginPrefix = '/' + pluginPrefix
     }
-    return R
+    return instancePrefix + pluginPrefix
```
Comment (allevo) [inline fastify.js]: What happen with `buildRoutePrefix('/foo/', 'bar)`? the returned string is `/foo//bar`

Comment (nwoltman) [reply fastify.js]: Yes. I think that's what the discussion in #584 is about.

**Comment by jsumners** [comment]
> Something about keeping the same shape of the fastify instance perhaps?

Exactly. I'd like to hear what @mcollina and/or @delvedor have to say before I add a LGTM.

**Comment by nwoltman** [comment]
Thanks for the clarification @delvedor :)
I added a new test with multiple `register` calls.

**Comment by mcollina** [comment]
Thanks!

**Pull request merged**